test/src/unit-QueryCondition.cc

```diff
@@ -742,8 +742,12 @@ void test_apply(const Datatype type, bool var_size, bool nullable) {
   test_apply_tile<T>(field_name, cells, type, &array_schema, &result_tile);
 }
 
+/* Note: Arithmetic operations cannot be performed on std::byte.
+    We will use uint8_t for the Datatype::BYTE case as it is the same size as
+   std::byte and can have arithmetic perfomed on it. */
 TEST_CASE("QueryCondition: Test apply", "[QueryCondition][apply]") {
   test_apply<int8_t>(Datatype::INT8);
+  test_apply<uint8_t>(Datatype::BYTE);
```

Comment:
I'm not sure we want this test. `Datatype::BYTE` is there to support blobs, as I recall. If that's the case, it should be treated as opaque. About the only condition that matters on it is equality.

Regardless, this highlights that we need stronger documentation about what this type is for.